### ISSUE TITLE
📚 Basic TypeScript types

### ISSUE THREAD

**Issue by scagood** [eslint-community/eslint-plugin-n#142]
We should probably look to add some super basic types I currently use:

```ts
declare module 'eslint-plugin-n' {
    import type { ESLint, Linter, Rule } from 'eslint';

    export default {} as {
        meta: {
            name: string;
            version: string;
        };
        rules: Record<string, Rule.RuleModule>;
        configs: {
            'recommended-module': ESLint.ConfigData
            'recommended-script': ESLint.ConfigData
            'recommended': ESLint.ConfigData
            'flat/recommended-module': Linter.FlatConfig
            'flat/recommended-script': Linter.FlatConfig
            'flat/recommended': Linter.FlatConfig
            'flat/mixed-esm-and-cjs': Linter.FlatConfig
        }
    };
}
```

**Comment by voxpelli** [comment]
Maybe we can generate this from JSDoc comments? That way they would also be validated against the current code and be a bit more DRY :)

**Comment by scagood** [comment]
mmm, I do prefer that idea, but it will be a fair bit more work as most of the doc comments dont really work currently :joy:

**Comment by voxpelli** [comment]
Maybe we can make it in two steps?

**Comment by scagood** [comment]
thinking about this, do we want to also try to have an auto release process?

Then we can build and version things in a more normalised manner.

I normally go with https://github.com/google-github-actions/release-please-action because it's simple to use
